src/System Application/App/File System/src/Account/FileAccountImpl.Codeunit.al

```diff
@@ -0,0 +1,227 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.FileSystem;
+
+using System.Text;
+using System.Utilities;
+
+codeunit 9451 "File Account Impl."
+{
+    Access = Internal;
+    InherentPermissions = X;
+    InherentEntitlements = X;
+    Permissions = tabledata "File System Connector Logo" = rimd,
+                  tabledata "File Scenario" = imd;
+
+    procedure GetAllAccounts(LoadLogos: Boolean; var TempFileAccount: Record "File Account" temporary)
+    var
+        FileAccounts: Record "File Account";
+        FileSystemConnector: Interface "File System Connector";
+        Connector: Enum "File System Connector";
+    begin
+        TempFileAccount.Reset();
+        TempFileAccount.DeleteAll();
+
+        foreach Connector in Connector.Ordinals do begin
+            FileSystemConnector := Connector;
+
+            FileAccounts.DeleteAll();
+            FileSystemConnector.GetAccounts(FileAccounts);
+
+            if FileAccounts.FindSet() then
+                repeat
+                    TempFileAccount := FileAccounts;
+                    TempFileAccount.Connector := Connector;
+
+                    if LoadLogos then
+                        ImportLogo(TempFileAccount, Connector);
+
+                    TempFileAccount.Insert();
+                until FileAccounts.Next() = 0;
+        end;
+
+        // Sort by account name
+        TempFileAccount.SetCurrentKey(Name);
+    end;
+
+    procedure DeleteAccounts(var FileAccountsToDelete: Record "File Account")
```

Comment:
See other comment about removing `temporary`, if its decided to keep the temporary for the variable definition, it should be added here, to align